confluence/client.py

```diff
@@ -515,6 +515,63 @@ def add_attachment(self, content_id, file_path, file_name=None, status=None):
                                               files={'file': (file_name, f)},
                                               data={})
 
+    def update_attachment(self,
+                          page_id,  # type: int
+                          attachment_id,  # type: int
+                          version,  # type: int
+                          new_filename=None,  # type: Optional[str]
+                          new_comment=None,  # type: Optional[str]
+                          new_media_type=None,  # type: Optional[str]
+                          new_page_id=None,  # type: Optional[int]
+                          new_status=None,  # type: Optional[ContentStatus]
+                          expand=None,  # type: Optional[List[str]]
+                          ):  # type: (...) -> Content
+        """
+        Update the information about an attachment in confluence.
+        This can be used to update filename, media type, comment
+        or status.
+
+        :param page_id: The parent page of the attachment.
+        :param attachment_id: Id of attachment to be updated
+        :param version: This should be the current version.
+        :param new_filename: The new filename to be used.
+        :param new_comment: The new comment to be used, optional.
+        :param new_media_type: The new comment to be used, optional.
+        :param new_page_id: The new page id for this attachment, optional.
+        :param new_status: The new content status, optional.
+        :param expand: An optional list of properties to be expanded on the resulting attachment object.
+
+        :return: The updated attachment object.
+        """
+        content = {
+            'id': attachment_id,
+            'type': ContentType.ATTACHMENT.value,
+            'version': {
+                'number': version,
+            },
+            'metadata': {}
+        }
+
+        if new_filename:
+            content['title'] = new_filename
+
+        if new_comment:
+            content['metadata']['comment'] = new_comment
+
+        if new_media_type:
+            content['metadata']['mediaType'] = new_media_type
+
+        if new_page_id:
```

Comment:
This section looks like it's worth a dedicated test to me, ensure that we're using the api how confluence expects (because I don't trust their docs at all any more)

Reply:
Added extra test